ui/app/src/Views/SilenceList/Views.elm

```diff
@@ -1,62 +1,76 @@
 import Time
 import Types exposing (Msg(UpdateFilter, MsgForSilenceList, Noop))
 import Views.FilterBar.Views as FilterBar
-import Views.FilterBar.Types as FilterBarTypes
+import Utils.String as StringUtils
 
 
 view : Model -> Time.Time -> Html Msg
 view model currentTime =
     case model.silences of
         Success sils ->
-            -- Add buttons at the top to filter Active/Pending/Expired
-            silences sils model.filterBar (text "")
+            div []
+                [ Html.map (MsgForFilterBar >> MsgForSilenceList) (FilterBar.view model.filterBar)
+                , a [ class "mb-4 btn btn-primary", href "#/silences/new" ] [ text "New Silence" ]
+                , silenceListView sils
+                ]
 
         Loading ->
             loading
 
         Failure msg ->
-            silences [] model.filterBar (error msg)
+            error msg
+
+
+silenceListView : List Silence -> Html Msg
+silenceListView silences =
+    div [] <|
+        List.map silenceGroupView <|
+            groupSilencesByState silences
 
 
-silences : List Silence -> FilterBarTypes.Model -> Html Msg -> Html Msg
-silences silences filterBar errorHtml =
+silenceGroupView : ( String, List Silence ) -> Html Msg
+silenceGroupView ( state, silences ) =
     let
-        html =
+        silencesView =
             if List.isEmpty silences then
-                div [ class "mt2" ] [ text "no silences found" ]
+                div [] [ text "No silences found" ]
             else
-                ul
-                    [ classList
-                        [ ( "list", True )
-                        , ( "pa0", True )
-                        ]
-                    ]
-                    (List.map silenceList silences)
+                ul [ class "list-group" ]
+                    (List.map silenceView silences)
     in
-        div []
-            [ Html.map (MsgForFilterBar >> MsgForSilenceList) (FilterBar.view filterBar)
-            , a [ class "f6 link br2 ba ph3 pv2 mr2 dib blue", href "#/silences/new" ] [ text "New Silence" ]
-            , errorHtml
-            , html
+        div [ class "mb-4" ]
+            [ h3 [] [ text <| StringUtils.capitalizeFirst state ]
+            , silencesView
             ]
 
 
-silenceList : Silence -> Html Msg
-silenceList silence =
+silenceView : Silence -> Html Msg
+silenceView silence =
     li
-        [ class "pa3 pa4-ns bb b--black-10" ]
+        [ class "list-group-item p-0 list-item" ]
         [ Views.Shared.SilenceBase.view silence
         ]
+
+
+groupSilencesByState : List Silence -> List ( String, List Silence )
+groupSilencesByState silences =
+    List.map (\state -> ( state, filterSilencesByState state silences )) states
+
+
+states : List String
+states =
+    [ "active", "pending", "expired" ]
+
+
+filterSilencesByState : String -> List Silence -> List Silence
+filterSilencesByState state =
+    List.filter (\{ status } -> status.state == state)
```

Comment:
This has a big-O value of `O(n^x)`, where `x == len(states)`. Depending on the number of silences this could be pretty painful. Could you either make a note to replace it with `groupBy` (`O(n)`) in this pr https://github.com/prometheus/alertmanager/pull/775/files#diff-4b0e2e6b80ca6bbf40b7a26e66b231b9R76, or just port it over and make a note to make sure we only have one of them when the above PR gets merged?